common/build-style/gemspec.sh

```diff
@@ -0,0 +1,131 @@
+	# Some even forget to update the version in the gemspec after releasing
+	sed -ri "s|(\.name .*)=.*|\1 = \"${pkgname#ruby-}\"|g" ${pkgname#ruby-}.gemspec
+	sed -ri "s|(\.version .*)=.*|\1 = \"${version}\"|g" ${pkgname#ruby-}.gemspec
+
+	# Replace use of `git ls-files` with find, use printf so we can print without starting
+	# dot-slash path
+	sed -i 's|`git ls-files`|`find . -type f -printf "%P\\n"`|g' ${pkgname#ruby-}.gemspec
+
+	# Sadly ruby isn't capable of handling nullbytes in a command so we have to use
+	# -print0, then try using sed to remove the suffix
+	# The end result is:
+	# `find . -type f -print0 | sed -e "s@\\./@@g"`
+	sed -i 's|`git ls-files -z`|`find . -type f -print0 \| sed -e "s@\\\\./@@g"`|g' ${pkgname#ruby-}.gemspec
+
+	if [ "$CROSS_BUILD" ]; then
+		# Create a .extconf file that forces the Makefile to use our environment
+		# this allows us to cross-compile like it is done with meson cross-files
+		cat>append<<EOF
+RbConfig::MAKEFILE_CONFIG['CPPFLAGS'] = ENV['CPPFLAGS'] if ENV['CPPFLAGS']
+\$CPPFLAGS = ENV['CPPFLAGS'] if ENV['CPPFLAGS']
+RbConfig::MAKEFILE_CONFIG['CC'] = ENV['CC'] if ENV['CC']
+RbConfig::MAKEFILE_CONFIG['CXX'] = ENV['CXX'] if ENV['CXX']
+RbConfig::MAKEFILE_CONFIG['LD'] = ENV['LD'] if ENV['LD']
+RbConfig::MAKEFILE_CONFIG['CFLAGS'] = ENV['CFLAGS'] if ENV['CFLAGS']
+RbConfig::MAKEFILE_CONFIG['CXXFLAGS'] = ENV['CXXFLAGS'] if ENV['CXXFLAGS']
+RbConfig::MAKEFILE_CONFIG['AR'] = ENV['AR'] if ENV['AR']
+RbConfig::MAKEFILE_CONFIG['RANLIB'] = ENV['RANLIB'] if ENV['RANLIB']
+EOF
+
+		# Patch all instances of extconf that use create_makefile
+		for f in $(find . -type f -name 'extconf.rb'); do
+			if [ ! -f ${f}.orig ]; then
+				# Ignore extconf files that do not create makefiles
+				grep -q create_makefile $f || continue
+				# Create a backup which we will restore later
+				cp $f ${f}.orig
+				# Patch extconf.rb for cross compile
+				cat append ${f}.orig >> $f
+			fi
+		done
+	fi
+
+	$gem_cmd build ${pkgname#ruby-}.gemspec
+
+	if [ "$CROSS_BUILD" ]; then
+		# Restore previous extconf.rb which we ship.
+		for f in $(find . -type f -name 'extconf.rb.orig'); do
+			mv $f ${f%.*}
+		done
+	fi
+}
+
+do_install() {
+	: ${gem_cmd:=gem}
+
+	local _GEMDIR _TARGET_PLATFORM _HOST_PLATFORM _HOST_EXT_DIR _TARGET_EXT_DIR
+
+	_GEMDIR="$($gem_cmd env gemdir)"
+
+	if [ "$CROSS_BUILD" ]; then
+		# Convert to the platforms used in our ruby packages
+		case "$XBPS_TARGET_MACHINE" in
+			x86_64) _TARGET_PLATFORM=x86_64-linux ;;
+			x86_64-musl) _TARGET_PLATFORM=x86_64-linux-musl ;;
+			i686) _TARGET_PLATFORM=i686-linux ;;
+			i686-musl) _TARGET_PLATFORM=i686-linux-musl ;;
+			armv7l) _TARGET_PLATFORM=armv7l-linux-eabihf ;;
+			armv6l) _TARGET_PLATFORM=arm-linux-eabihf ;;
+			armv7l-musl) _TARGET_PLATFORM=armv7l-linux-musleabihf ;;
+			armv6l-musl) _TARGET_PLATFORM=arm-linux-musleabihf ;;
+			aarch64) _TARGET_PLATFORM=aarch64-linux ;;
+			aarch64-musl) _TARGET_PLATFORM=aarch64-linux-musl ;;
+		esac
```

Comment:
What about armv{6,7}hf? I can't keep up with all of those arm variants.

Reply:
No those are the names ruby picked when being compiled on Void and if you want to change and make them the same them modify the `ruby` package to do so.

Reply:
Eh? I'm talking about handling armv6/7hf to the case statement